File: sprint2/kotlin/lesson5code1.kt
class Node(var value: Any? = null, var next: Node? = null)

fun printLinkedList(vertex: Node?) {
    var current = vertex
    while (current != null) {
        print("${current.value} -> ")
        current = current.next
    }
    println("None")
}


val n3 = Node(value = 'third')
val n2 = Node(value = 'second', next = n3)
val n1 = Node(value = 'first', next = n2)
printLinkedList(n1)
>>> first -> second -> third -> None

printLinkedList(n2)
>>> second -> third -> None

File: sprint2/By languages/C++/lesson5code1.cpp
#include <iostream>

struct Node {
    int value;
    Node* next = nullptr;
}

void print_linked_list(Node* vertex) {
    while (vertex) {
        std::cout << vertex->value << " -> ";
        vertex = vertex->next;
    }
    std::cout << "None" << std::endl;
}

int main() {
    Node* n3 = new Node("third");
    Node* n2 = new Node("second", n3);
    Node* n1 = new Node("first", n2);
    print_linked_list(n1);
    print_linked_list(n2);

    delete n3;
    delete n2;
    delete n1;
    return 0;
}